Tensile/Common.py

```diff
@@ -2310,6 +2318,45 @@ def which(p):
                 return candidate
     return None
 
+
+def populateCapabilities(
+    globalParameters: Dict[str, Any], cachedAsmCaps: Dict[IsaVersion, dict]
+):
+    """Populates the assembler and archiecture capabilities based on the compiler and ISA.
+
+    This function updates the **globalParameters** and **cachedAsmCaps** dictionaries with
+    the assembler and architecture capabilities for each supported ISA version. It checks
+    the compatibility of each ISA version with the current ROCm stack version and skips any
+    unsupported ISA versions, issuing a warning for each skipped version.
+
+    Args:
+        globalParameters: A dictionary containing global parameters of the
+            application, including the HipClang version and supported ISA versions.
+        cachedAsmCaps: A dictionary to be populated with the assembler
+            capabilities for each ISA version.
+
+    Note:
+        This function modifies `globalParameters` and `cachedAsmCaps` in place.
+    """
+    compilerVersion = CompilerVersion(
+        *[int(c) for c in globalParameters["HipClangVersion"].split(".")[:2]]
+    )
+    supportedISA = globalParameters["SupportedISA"]
+    to_remove = []
+
+    for v in supportedISA + [(0, 0, 0)]:
+        if v[0] == 12 and not (compilerVersion.major == 6 and compilerVersion.minor == 3):
```

Comment:
Should this be like 
  not (major > 6 or (major ==6 and minor >=3))

Reply:
I agree, I will update this to include all compiler versions 6.3 and up.